config/default/CompatibilityChecker.conf.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 Open Assessment Technologies SA
+ */
+return new oat\taoClientDiagnostic\model\CompatibilityChecker();
```

Comment:
just a note, if there's a service without public config options, the config file may be omitted as well as SERVICE_ID and a migration to register it. We could simply extend `ServiceConfigurable` and read the service from ServiceLocator by `::class`:

`$this->getServiceLocator()->get(CompatibilityChecker::class);`

that's just for info, no need to refactor what's done already in this PR